src/Symfony/Component/Filesystem/Tests/FilesystemTest.php

```diff
@@ -1102,6 +1102,37 @@ public function testDumpKeepsExistingPermissionsWhenOverwritingAnExistingFile()
         $this->assertFilePermissions(745, $filename);
     }
 
+    /**
+     * @expectedException \Symfony\Component\Filesystem\Exception\IOException
+     * @expectedExceptionMessageRegExp /^The parent path of ".*" is a file\.$/
+     */
+    public function testDumpFailsIfParentDirectoryIsAnExistingFile()
+    {
+        $file = $this->workspace.DIRECTORY_SEPARATOR.'foo';
+        $target = $file.DIRECTORY_SEPARATOR.'bar';
+        touch($file);
+
+        $this->filesystem->dumpFile($target, 'baz');
+    }
+
+    /**
+     * @expectedException \Symfony\Component\Filesystem\Exception\IOException
+     * @expectedExceptionMessageRegExp /^The parent path of ".*" is a symlink to a nonexistent directory\.$/
+     */
+    public function testDumpFailsIfTargetIsSymlinkAndSymlinkTargetDoesNotExist()
+    {
+        $this->markAsSkippedIfSymlinkIsMissing();
```

Comment:
We need to check why this does not apply on Appveyor or if `is_link()` works differently on Windows than on Unix. If someone using Windows can help here, that would be really appreciated.